cmd/system-probe/subcommands/run/command.go

```diff
@@ -69,13 +71,15 @@ func Commands(globalParams *command.GlobalParams) []*cobra.Command {
 		RunE: func(cmd *cobra.Command, args []string) error {
 			return fxutil.OneShot(run,
 				fx.Supply(cliParams),
-				fx.Supply(config.NewAgentParamsWithoutSecrets("", config.WithConfigMissingOK(true))),
+				fx.Supply(config.NewAgentParams("", config.WithConfigMissingOK(true))),
 				fx.Supply(sysprobeconfigimpl.NewParams(sysprobeconfigimpl.WithSysProbeConfFilePath(globalParams.ConfFilePath))),
+				fx.Supply(secrets.NewDisabledParams()),
```

Comment:
I think this was here before I was using the MockModule, so a Params needed to be Provide'd, but it's not necessary now.

Reply:
As we tend to copy/paste setup we should, when possible, keep it to the bare minimum. So if we can remove it let's do that.

Reply:
Removing this was making a test fail, but the correct fix was actually make the secrets.Component optional using the struct tag, as you had suggested. Now there's no need to include a secrets to satisfy the config, and not providing it will just skip secret decryption.